napari/layers/_data_protocols.py

```diff
@@ -0,0 +1,129 @@
+"""This module holds Protocols that layer.data objects are expected to provide.
+"""
+from __future__ import annotations
+
+from types import GeneratorType
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    TypeVar,
+    Union,
+)
+
+import numpy as np
+from typing_extensions import Protocol, runtime_checkable
+
+from .utils.layer_utils import compute_multiscale_level_and_corners
+
+_T = TypeVar('_T')
+Shape = Tuple[int, ...]
```

Comment:
I came back to this definition when I saw `Tuple[Shape]` below, maybe `ArrayShape` would be more self-evident/less ambiguous?